asf_public_discourse_home_decarbonisation/utils/text_processing_utils.py

```diff
@@ -0,0 +1,101 @@
+import re
+
+
+def remove_urls(text: str) -> str:
+    """Remove URLs from text.
+
+    Args:
+        text (str): text to process
+
+    Returns:
+        str: text with URLs removed
+    """
+    return re.sub(r"http\S+", " ", text)
+
+
+def remove_username_pattern(text: str) -> str:
+    """
+    Remove the pattern "username wrote:" from text.
+
+    Args:
+        text (str): text to process
+
+    Returns:
+        str: text with the pattern removed
+    """
+    # Define the pattern to match "username wrote:" and remove it
+    pattern = re.compile(r"\w+\s+wrote: »\n")
+    cleaned_text = re.sub(pattern, " ", text)
+
+    return cleaned_text
+
+
+def remove_introduction_patterns(text: str) -> str:
+    """Remove introduction patterns from text.
+
+    Args:
+        text (str): text to process
+
+    Returns:
+        str: text with introduction patterns removed
+    """
+
+    # Define the patterns to match and remove
+    patterns = [
+        r"\b(hi|hello|heya|hey)\s+(everyone|all|there)?[!,.]?\s*\n?",
+        r"\b(hi|hello|heya|hey)[!,.]?\s*\n?",
+    ]
+
+    # Combine patterns into a single regex pattern
+    combined_pattern = "|".join(patterns)
+
+    # Compile the combined pattern
+    pattern = re.compile(combined_pattern, re.IGNORECASE)
+
+    # Remove patterns from the text
+    cleaned_text = re.sub(pattern, " ", text)
+
+    return cleaned_text
+
+
+def process_abbreviations(text: str) -> str:
```

Comment:
I've added this function to the utils, as we use it accross different analysis scripts, and it's important that we always use the exact same transformation.

I removed "hp" and "hps" as it was replacing ".php" with heat pumps, which is obviously wrong.

To be honest, the way we should be doing this is, we would use spacy to tokenise, and then apply the transformation to the tokens. Afterwards, we would put the tokens back together to form the sentence, including punctuation. But I think this is fine for now?

Reply:
The results seem clean for the most part so happy to stick with this for now :)